standard/gallery.lua

```diff
@@ -0,0 +1,118 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Gallery
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local Array = require('Module:Array')
+local Image = require('Module:Image')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+
+local DEFAULT_HEIGHT = 200
+
+local Gallery = {}
+
+function Gallery.template(frame)
+	return Gallery.run(Arguments.getArgs(frame))
+end
+
+---@param args table?
+---@return Html?
+function Gallery.run(args)
+	args = args or {}
+
+	local height = tonumber((string.gsub(args.height or '', 'px$', ''))) or DEFAULT_HEIGHT
+
+	---@param index integer
+	---@return {imageLightMode: string, imageDarkMode: string?, caption: string, below: string|Html?, link: string?}?
+	local processImageInput = function(index)
+		local rawInput = args[index]
+		local input = type(rawInput) == 'table' and rawInput or Json.parseIfTable(args[index])
+		if Logic.isEmpty(input) then return end
+		---@cast input -nil
+
+		local imageLightMode = input.lightmode or input[1]
+		if Logic.isEmpty(imageLightMode) then return end
+
+		return {
+			imageLightMode = imageLightMode,
+			imageDarkMode = input.darkmode,
+			caption = input.caption,
+			link = input.link,
+			below = input.below,
+		}
+	end
+
+	local images = Array.mapIndexes(processImageInput)
+
+	if Logic.isEmpty(images) then return end
+
+	local gallery = mw.html.create('ul')
+		:addClass('gallery mw-gallery-packed')
```

Comment:
it is not as easy as changing the class used here

would need a different display function for each mode

since from what i see oin the wikis mostly `packed` is used i think we start with it and leave the other modes oos for now

alternative would be to use `mw.getCurrentFrame():callParserFunction(gallery, {...})`
but then we would need to do a gallery for light and one for darkmode (wrapping them in divs with `show-when-light-mode` or `show-when-dark-mode` respectively